dq.go

```diff
@@ -0,0 +1,212 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/TykTechnologies/tyk-cluster-framework/client"
+	"github.com/TykTechnologies/tyk-cluster-framework/encoding"
+
+	"github.com/jeffail/tunny"
+
+	"github.com/Sirupsen/logrus"
+
+	"github.com/TykTechnologies/dq"
+	"github.com/TykTechnologies/tyk/apidef"
+)
+
+var DQFlusherPool *tunny.WorkPool = tunny.CreatePoolGeneric(10)
+var QuotaHandler *dq.DistributedQuota
+
+type GetLeaderStatusFunc func() bool
+
+func getDQTopic() string {
+	topic := "tyk.dq"
+	if config.DBAppConfOptions.NodeIsSegmented {
+		if len(config.DBAppConfOptions.Tags) > 0 {
+			tags := strings.Join(config.DBAppConfOptions.Tags, ".")
+			topic = topic + "." + tags
+		}
+	}
+
+	return topic
+}
+
+func dqErrorHandler(e error) {
+	log.WithFields(logrus.Fields{
+		"prefix": "main.DQ",
+	}).Error(e)
+}
+
+var dummyAPISpec = APISpec{APIDefinition: &apidef.APIDefinition{SessionLifetime: 0}}
+
+func dqFlusher(d map[string]*dq.Quota) error {
+	for k, v := range d {
+		DQFlusherPool.SendWork(func() {
+			// Ignore deleted flags
+			if v.Delete {
+				log.Warning("Key has been tagged for deletion, not writing counter")
+				return
+			}
+
+			// We will track all the session handlers for this key
+			processedSpecs := map[SessionHandler]struct{}{}
+
+			// Let's go through all the API IDs in the metadata so we capture all the handlers
+			apis := v.Meta.(map[string]interface{})["Apis"]
+
+			// I hate this
+			var exp int64
+			switch x := v.Meta.(map[string]interface{})["QuotaRenewal"].(type) {
+			case int64:
+				exp = x
+			case float64:
+				exp = int64(x)
+			}
+
+			expT := time.Unix(exp, 0)
+			for _, aid := range apis.([]interface{}) {
+				apiID := aid.(string)
+
+				// This will grab the session handler
+				spec := GetSpecForApi(apiID)
+				if spec == nil {
+					log.Warning("Can't find back-end for this API, skippingc")
+					break
+				}
+
+				// Have we processed on this handler before (many APIs may use the same handler)?
+				_, processedOnSH := processedSpecs[spec.SessionManager]
+				if processedOnSH {
+					continue
+				}
+
+				// This handler hasn't been used yet for the API
+				// Get the session data
+				s, f := spec.SessionManager.GetSessionDetail(k)
+
+				// If it was found, lets process it for this handler
+				if !f {
+					// No longer in session store, delete
+					QuotaHandler.TagDelete(k)
+					continue
+				}
+
+				skip := false
+				if time.Now().After(expT) {
+					QuotaHandler.TagDelete(k)
+					skip = true
+				}
+
+				if s.IsExpired() {
+					// Remove expired data too
+					QuotaHandler.TagDelete(k)
+					skip = true
+				}
+
+				if !skip {
```

Comment:
You can get rid of this `if` and just use `continue` in all the `skip = true` lines.